src/pudl/transform/eia861.py

```diff
@@ -661,6 +679,84 @@ def _compare_totals(data_cols, idx_cols, class_type, df_name):
                 f'{df_name}: for column {col} all total values are NaN')
 
 
+def _clean_nerc_add_rows(df, idx_cols, idx_no_nerc):
+    """Clean NERC region entries and make new rows for multiple nercs.
+
+    This function examines reported NERC regions and makes sure the output column of
+    the same name has reliable, singular NERC region acronyms. To do so, this function
+    identifies entries where there are two or more NERC regions specified in a single cell
+    (such as SPP & ERCOT) and makes new, duplicate rows for each NERC region. It also
+    converts non-recognized reported nerc regions to 'UNK'.
+
+    Args:
+        df (pandas.DataFrame): A DataFrame with the column 'nerc_region' to be
+            cleaned.
+        idx_cols (list): A list of the primary keys.
+        idx_no_nerc (list): A list of the primary keys, not including nerc_region.
+
+    Returns:
+        pandas.DataFrame: A DataFrame with correct and clean nerc regions.
+
+    """
+    # Split raw df into primary keys plus nerc region and other value cols
+    nerc_df = df[idx_cols]
+    other_df = df.drop('nerc_region', axis=1).set_index(idx_no_nerc)
+
+    # Make all values upper-case
+    # Replace all NA values with UNK
+    # Make nerc values into lists to see how many separate values are stuffed into one row (ex: SPP & ERCOT)
+    nerc_df = (
+        nerc_df.assign(
+            nerc_region=(lambda x: (
+                x.nerc_region
+                .str.upper()
+                .fillna('UNK')
+                .str.findall(r'[A-Z]+'))),
+            multiple_nercs=(lambda x: (
+                x.nerc_region.apply(lambda x: len(x) > 1)))
+        )
+    )
+
+    # Needed nerc_regions as a list to calculate multiple_nercs, but now can convert nerc lists into
+    # comma-separated strings (can't use same col name twice in one assign)
+    nerc_df['nerc_region'] = nerc_df['nerc_region'].str.join(',')
+
+    # Make dfs of multiple vs. single nerc values
+    nerc_multiples = (
+        nerc_df[nerc_df['multiple_nercs']].copy()
+        .drop('multiple_nercs', axis=1)
+    )
+    nerc_singles = (
+        nerc_df[~nerc_df['multiple_nercs']].copy()
+        .drop('multiple_nercs', axis=1)
+    )
+
+    new_nerc_rows = (
+        nerc_multiples
+        .set_index(idx_no_nerc)
+        .stack()
+        .str.split(',', expand=True)
+        .stack()
+        .unstack(-2)
+        .reset_index(-1, drop=True)
+        .reset_index()
+    )
```

Comment:
My one fear about this is that we are going to end up having multiple rows with different nerc regions but the same data. Although I'm not sure what the best thing would be here :-/

Do you know the scale of the problem here? like how many `nerc_multiples` exist? How many unique bad strings/combos are there? If it is not too many we could flag, replace or null them. We should at least think through how to deal with these duplicate records if we keep them.

Reply:
Luckily, _most_ instances of multiple entires in a single cell are comprised of a legitimate nerc region paired with a non legitimate nerc region. I don't have exact numbers, but essentially what this means is that we go through all the effort of splitting them apart only to delete the "fake" nerc row.

Reply:
Instead of generating these new records it might be better to break the records apart like you are currently, drop or clean the non-recognized codes, and when there are still multiple recognizable codes squish them back together in a standard way. like sort the multiple codes so they are all in the same order and squish them back into one string separated by `-` or `_`. This way you could break them apart easily and we would have a smaller number of these weird combo codes.

Reply:
Will do!

Reply:
One thing I was thinking about when I made this the first time around was whether I should keep at least a placeholder for the weird, non-nerc regions listed. In the first iteration I changed them to UNK. Do you think I should just drop them? I'll keep a copy of the old function in case we want to keep tweaking it.